lib/Sema/ConstraintSystem.cpp

```diff
@@ -486,8 +486,25 @@ Type ConstraintSystem::openUnboundGenericType(
        ConstraintLocatorBuilder locator) {
   assert(!type->hasTypeParameter());
 
-  if (!type->hasUnboundGenericType())
+  // If this is a generic typealias defined inside of constrainted extension,
+  // let's add all of the generic requirements to the constraint system to
+  // make sure that it's something we can use.
+  if (auto *NAT = dyn_cast<NameAliasType>(type.getPointer())) {
+    auto *decl = NAT->getDecl();
+    auto *extension = dyn_cast<ExtensionDecl>(decl->getDeclContext());
+    if (extension && extension->isConstrainedExtension()) {
+      auto substitutions = NAT->getSubstitutionMap();
+      if (auto *signature = substitutions.getGenericSignature())
+        openGenericRequirements(
+            decl->getDeclContext(), signature,
```

Comment:
Nit: You could just pass `extension` here.

Reply:
Done!